lld/ELF/LinkerScript.h

```diff
@@ -161,8 +264,12 @@ struct MemoryRegion {
   uint32_t negInvFlags;
   uint64_t curPos = 0;
 
-  uint64_t getOrigin() const { return origin().getValue(); }
-  uint64_t getLength() const { return length().getValue(); }
+  uint64_t getOrigin() const {
```

Comment:
Is it a precondition that origin be non-null if getOrigin() is called? If so it could be worth doing an assert(origin != nullptr) instead of picking a default value of 0.

Same for length below.

At a glance it looks like origin and length are passed in via the constructor and it looks like they can't be nullptr.

Reply:
Thank you for the review! I removed the precondition and added assert for `origin` and `length`. Yes I think they should not be `nullptr` since they are passed in the constructor but still have to make sure that they are not passed in as `nullptr` so an assert would be perfect here.